WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt

```diff
@@ -517,13 +518,21 @@ class MySiteFragment : Fragment(R.layout.my_site_fragment),
     }
 
     private fun MySiteFragmentBinding.loadData(cardAndItems: List<MySiteCardAndItem>) {
+        if(actionableEmptyView.isVisible) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.MY_SITE_NO_SITES_VIEW_HIDDEN)
```

Comment:
Suggestion (💡): Consider moving this logic to the `ViewModel` so that it can be tested. On Android we are usually pushing all logic, included tracking, into this layer in order to have the upper view layer be as dump as possible.

Also, by doing so you can make this whole `actionableEmptyView` logic atomic. What I mean by that is that you can make sure that the `NO_SITES` tracking will always happen before the `actionableEmptyView` changes its visibility state, and this way you can guaranteed that if someone reorder the lines on the view layer, on `loadData(...)` and/or `loadEmptyView(...)`, the result will remain the same.

As an example, consider the below solution:

`MyStieFragment.kt`
```
    private fun MySiteFragmentBinding.loadData(cardAndItems: List<MySiteCardAndItem>) {
        recyclerView.setVisible(true)
        viewModel.setActionableEmptyViewGone(actionableEmptyView.isVisible) {
            actionableEmptyView.setVisible(false)
        }
        swipeToRefreshHelper.isRefreshing = false
        (recyclerView.adapter as? MySiteAdapter)?.loadData(cardAndItems)
    }

    private fun MySiteFragmentBinding.loadEmptyView(shouldShowEmptyViewImage: Boolean) {
        recyclerView.setVisible(false)
        viewModel.setActionableEmptyViewVisible(actionableEmptyView.isVisible) {
            actionableEmptyView.setVisible(true)
            actionableEmptyView.image.setVisible(shouldShowEmptyViewImage)
        }
        swipeToRefreshHelper.isRefreshing = false
    }
```

`MySiteViewModel.kt`
```
    fun setActionableEmptyViewGone(isVisible: Boolean, setGone: () -> Unit) {
        if (isVisible) analyticsTrackerWrapper.track(Stat.MY_SITE_NO_SITES_VIEW_HIDDEN)
        setGone()
    }

    fun setActionableEmptyViewVisible(isVisible: Boolean, setVisible: () -> Unit) {
        if (!isVisible) analyticsTrackerWrapper.track(Stat.MY_SITE_NO_SITES_VIEW_DISPLAYED)
        setVisible()
    }
```

Now, on the view layer, it is guaranteed that the tracking will take place before the visibility update, and on the view model layer, we can unit test the solution. Wdyt?

Reply:
@ParaskP7 ooo! This idea is better than what I had. Thanks for helping me out with this!

Reply:
Thank YOU @emilylaguna for even considering it! 🚀